arrow/arrow.py

```diff
@@ -9,84 +9,90 @@ def choose(n, k):
     return np.rint(product)
 
 
-def propensity(reaction, state, form):
-    reactants = np.where(reaction < 0)
+def propensity(stoichiometry, state, form):
+    reactants = np.where(stoichiometry < 0)
     terms = [
-        form(state[reactant], -reaction[reactant])
+        form(state[reactant], -stoichiometry[reactant])
         for reactant in reactants[0]]
 
     return np.array(terms).prod()
 
-
-def step(reactions, rates, state, forms, propensities=[], update_reactions=()):
+def step(stoichiometry, rates, state, forms, propensities=[], update_reactions=()):
     if len(update_reactions):
         for update in update_reactions:
-            reaction = reactions[update]
+            reaction_stoichoiometry = stoichiometry[:, update]
```

Comment:
My only issue with this whole `reaction_stoichiometry` vs `stoichiometry` vs `reaction` business is that if we are using `reaction_stoichiometry` then the other `stoichiometry` should be comparably qualified, perhaps `system_stoichiometry`, to make it conceptually consistent. However, that gets cumbersome everywhere. 

My other issue is that `reaction_stoichiometry` is long : ) I know `reaction` as a shorthand for this concept is perhaps not exactly precise, but I think it makes sense in context here. Also, not so into the word `stoichiometry` standing in for both all reactions and a single reaction. Any other alternatives here?

Reply:
Yeah, the language is ugly.  After this pass I'm inclined to go back and distinguish the `stoichiometric_matrix` (the full system) versus a `stoichiometry` (which is reaction specific).  The former is verbose but explicit.

Reply:
Hmmm that seems like a good middle ground.